vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/batchRegistration/SampleSpreadsheetLayout.java

```diff
@@ -107,83 +169,303 @@ private static class SampleRegistrationSheetBuilder {
+      } else {
+        dropdownCellFactory.addDropDownCell(rowIndex, colIndex);
+      }
+    }
+
     public void addSheetToSpreadsheet(MetaDataTypes metaDataTypes, Spreadsheet spreadsheet) {
-      spreadsheet.setActiveSheetProtected("pwd");
+      spreadsheet.setActiveSheetProtected("password-needed-to-lock");
+      dropdownCellFactory = new SpreadsheetDropdownFactory();
+
       switch (metaDataTypes) {
-        case PROTEOMICS -> addProteomicsSheet(spreadsheet);
-        case LIGANDOMICS -> addLigandomicsSheet(spreadsheet);
-        case TRANSCRIPTOMICS_GENOMICS -> addGenomicsSheet(spreadsheet);
-        case METABOLOMICS -> addMetabolomicsSheet(spreadsheet);
+        case PROTEOMICS -> addProteomicsSheet(spreadsheet, sampleRegistrationService.retrieveProteomics());
+        case LIGANDOMICS -> addLigandomicsSheet(spreadsheet, sampleRegistrationService.retrieveLigandomics());
+        case TRANSCRIPTOMICS_GENOMICS -> addGenomicsSheet(spreadsheet, sampleRegistrationService.retrieveGenomics());
+        case METABOLOMICS -> addMetabolomicsSheet(spreadsheet, sampleRegistrationService.retrieveMetabolomics());
+      }
+      spreadsheet.setSpreadsheetComponentFactory(dropdownCellFactory);
+      //initialise first rows based on known sample size
+      for(int rowIndex = 1; rowIndex <= numberOfSamples; rowIndex++) {
+        addRow(spreadsheet);
       }
     }
 
-    private void unlockColumn(Spreadsheet spreadsheet, int column, int minRow, int maxRow) {
-      List<Cell> updatedCells = new ArrayList<Cell>();
+    private void setupCommonDropDownColumns() {
+      initDropDownColumn(header.indexOf("Species"), species);
+      initDropDownColumn(header.indexOf("Specimen"), specimens);
+      initDropDownColumn(header.indexOf("Condition"), conditionsToReplicates.keySet().stream().toList());
+      initDropDownColumn(header.indexOf("Biological replicate id"), getReplicateLabels());
+    }
+
+    private void initDropDownColumn(int colIndex, List<String> items) {
+      if(items.size() > 1) {
+        DropDownColumn itemDropDown = new DropDownColumn();
+        itemDropDown.withItems(items);
+        itemDropDown.toRowIndex(0).atColIndex(colIndex);
+        dropdownCellFactory.addDropdownColumn(itemDropDown);
+      }
+    }
+
+    private void unlockEmptyCells(Spreadsheet spreadsheet) {
+      for(int column = 0; column < Integer.MAX_VALUE; column++) {
+        Cell firstCell = spreadsheet.getCell(0, column);
+        Cell firstDataCell = spreadsheet.getCell(1, column);
+        boolean hasHeader = !isCellEmpty(firstCell);
+        if(!hasHeader) {
+          break;
+        }
+        boolean hasData = !isCellEmpty(firstDataCell);
+        boolean hasDropdown = dropdownCellFactory.findColumnInRange(1, column)!=null;
+        //columns need to be unlocked if they have a header and no data/dropdown
+        if(!hasData && !hasDropdown) {
+          unlockCell(spreadsheet, column, 1);
+        }
+      }
+    }
```

Comment:
NIT: I think this could also be simplified if we specify the unlocked columns beforehand